dbms/src/Common/ProcessCollector.h

```diff
@@ -0,0 +1,49 @@
+// Copyright 2023 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <ProcessMetrics/ProcessMetrics.h>
+#include <prometheus/counter.h>
+#include <prometheus/metric_family.h>
+#include <prometheus/registry.h>
+
+#include <mutex>
+
+namespace DB
+{
+
+class ProcessCollector : public prometheus::Collectable
```

Comment:
Add some comments to clarify the difference between registry？

Reply:
done